### DISCUSSION TITLE
Is it possible to group-wise color every other row?

### DISCUSSION THREAD

**Discussion by AlbertRapp** [rstudio/gt#1321] [Q&A]
In this table I have colored every other row.

```
library(gt)
dat <- towny |> 
  select(name, status, land_area_km2) |> 
  slice_sample(n = 3, by = status)

dat |> 
  gt() |> 
  tab_style(
    style = cell_fill(color = 'grey90'),
    locations = list(
      cells_body(rows = seq(1, nrow(dat), 2))
    )
  )
```

![image](https://user-images.githubusercontent.com/65388595/236309589-322f0a62-29a9-4a8e-816d-c0676dbe3dbc.png)

However, is it also possible to achieve every other row on a per-group basis? With the following code, the coloring is different for the different groups.

```
dat |> 
  arrange(status) |> 
  gt(groupname_col = 'status') |>
  tab_style(
    style = cell_fill(color = 'grey90'),
    locations = list(
      cells_body(rows = seq(1, nrow(dat), 2))
    )
  ) |> 
  tab_style( ## Only added to see groups better
    style = cell_text(
      color = 'dodgerblue4', weight = 'bold', font = google_font('Merriweather')
    ),
    locations = cells_row_groups()
  )
```

![image](https://user-images.githubusercontent.com/65388595/236309863-f4c216e6-4d68-478b-93b7-0a0f8b593999.png)

So in the desired row I'd like to see 'Nairn and Hyman' and 'Pelee' have a grey background. Is that possible?

**Comment by eipi10** [comment]
I can't find a way to color by group within the natural `gt` workflow, but below is a function that provides a workaround.

```r
library(tidyverse)
library(gt)

color_by_group = function(x, ...) {
  
  x = x %>% 
    ungroup %>% 
    arrange(...)
  
  myrows = x %>% 
    mutate(row.num = 1:n()) %>% 
    group_by(...) %>% 
    mutate(keep=1:n() %% 2 == 1) %>% 
    ungroup() %>% 
    filter(keep) %>% 
    pull(row.num)
  
  group.col.names = enquos(...) %>% map_chr(as_label)
  
  x %>%
    gt(groupname_col = group.col.names) %>%
    tab_style(
      style = cell_fill(color = 'grey90'),
      locations = list(
        cells_body(rows = myrows)
      )
    ) 
}

set.seed(2)
dat = towny %>% 
   select(name, status, land_area_km2) %>% 
   slice_sample(n = 3, by = status)

dat %>% 
  color_by_group(status)
```

![image](https://user-images.githubusercontent.com/436817/236649031-5d6d3d07-1433-4adf-a490-b12daa04730c.png)

```r
mtcars %>% 
  slice(10:15) %>% 
  select(cyl, carb, mpg, hp) %>% 
  color_by_group(cyl, carb) 
```
![image](https://user-images.githubusercontent.com/436817/236649256-ffdd3fd8-7437-44d3-bad5-140a2c72ca06.png)

**Comment by rich-iannone** [comment]
@eipi10 excellent function! On this note, targeting is pretty weak when groups are involved. I think there needs to be things like a `group` argument in functions where there is already a `rows` argument. Or, some specialized syntax or helper functions that could be used in `rows` (an `every_2()` helper function that's group-aware would do the trick for the problem at hand). I will eventually get some thoughts together and put them in a new issue, but yeah, this part of gt needs some enhancement.
